lib/api/BackbeatAPI.js

```diff
@@ -401,6 +402,72 @@ class BackbeatAPI {
         });
     }
 
+    /**
+     * For the given queue enry's site, send an entry with PENDING status to the
+     * replication status topic, then send an entry to the replication so that
+     * the queue processor re-attempts replication.
+     * @param {QueueEntry} queueEntry - The queue entry constructed from the
+     * failed kafka entry that was stored as a Redis hash value.
+     * @param {Function} cb - The callback.
+     * @return {undefined}
+     */
+    _pushKafkaTopics(queueEntry, cb) {
+        const site = queueEntry.getSite();
+        const pendingEntry = queueEntry.toPendingEntry(site);
+        const retryEntry = queueEntry.toRetryEntry(site);
+        // Send messages in series to ensure status is updated in correct order.
```

Comment:
The guarantee applies in the sense that: if you push multiple entries pertaining to *the same object* and in *the same queue*, they will be received in order by the same consumer.

That's it, there's no ordering guarantee when you push to different queues for different kind of processing, although it applies to the same object, because they will end up in different partitions and be processed by different consumers (and different processes in this case).

Reply:
Okay, I think this ordering should apply then since the object MD is for the same object. The replication status processor will first get the PENDING status for the object, then it will get whatever status is set on the retry of the operation (which could be, for example, 'FAILED' or 'COMPLETED').